homeassistant/components/dsmr/sensor.py

```diff
@@ -82,10 +83,12 @@ def update_entities_telegram(telegram: dict[str, DSMRObject]) -> None:
 
     # Creates an asyncio.Protocol factory for reading DSMR telegrams from
     # serial and calls update_entities_telegram to update entities on arrival
-    is_rfxtrx = entry.data.get(CONF_IS_RFXTRX)
+    protocol = entry.data.get(CONF_PROTOCOL, DSMR_PROTOCOL)
     if CONF_HOST in entry.data:
         reader_factory = partial(
-            create_tcp_dsmr_reader if not is_rfxtrx else create_rfxtrx_tcp_dsmr_reader,
+            create_tcp_dsmr_reader
```

Comment:
We prefer to avoid ternary operator expression if it spans more than one line.

Reply:
Changed it into a separate if/else.